blackbox/docs/appendices/release-notes/3.0.3.rst

```diff
@@ -0,0 +1,95 @@
+.. _version_3.0.3:
+
+=============
+Version 3.0.3
+=============
+
+Released on 2018/06/29.
+
+.. NOTE::
+
+   If you are upgrading a cluster, you must be running CrateDB 2.0.4 or higher
+   before you upgrade to 3.0.3.
+
+   We recommend that you upgrade to the latest 2.3 release before moving to
+   3.0.3.
+
+   You cannot perform a `rolling upgrade`_ to this version. Any upgrade to this
+   version will require a `full restart upgrade`_.
+
+   When restarting, CrateDB will migrate indexes to a newer format. Depending
+   on the amount of data, this may delay node start-up time.
+
+   Please consult the :ref:`version_3.0.0_upgrade_notes` before upgrading.
+
+.. WARNING::
+
+    Tables that were created prior to upgrading to CrateDB 2.x will not
+    function with 3.0 and must be recreated before moving to 3.0.x.
+
+    You can recreate tables using ``COPY TO`` and ``COPY FROM`` while running a
+    2.x release into a new table, or by `inserting the data into a new table`_.
+
+.. WARNING::
+
+   Before upgrading, you should `back up your data`_.
+
+.. _rolling upgrade: http://crate.io/docs/crate/guide/best_practices/rolling_upgrade.html
+.. _full restart upgrade: http://crate.io/docs/crate/guide/best_practices/full_restart_upgrade.html
+.. _back up your data: https://crate.io/a/backing-up-and-restoring-crate/
+.. _inserting the data into a new table: https://crate.io/docs/crate/reference/en/latest/admin/system-information.html#tables-need-to-be-recreated
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Changelog
+=========
+
+Changes
+-------
+
+- Added settings ``s3.client.default.access_key`` and
+  ``s3.client.default.secret_key`` which can be used to set default credentials
+  for s3 repositories, if they are not passed as parameters to the
+  ``CREATE REPOSITORY`` SQL statement.
+
+- Implemented a thread-utilization down-scaling logic which dynamically adapts
+  the number of threads used for ``SELECT`` queries to avoid running into
+  ``RejectedExcecution`` errors if there are many shards per node involved in
+  the queries.
+
+- Changed internals for DELETE and UPDATE queries which should generally result
+  in a performance increase for queries which only match a subset of the rows
```

Comment:
just a personal preference, but I'd reword this to remove "internals". Just saying we improved performance in these cases would be fine & reads better